docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
-[example configuration](https://github.com/DeepRegNet/DeepReg/blob/main/config/unpaired_labeled_ddf.yaml),
-with comments on the available options in each subsection.
+subsections, `method`, `backbone`, `loss`, `optimizer`, `preprocess` and other training
+hyper-parameters, including `epochs` and `save_period`.
+
+### Method - required
+
+The `method` argument defines the registration type. It must be a string type, one of
+"ddf", "dvf", "conditional", which are the currently supported registration methods.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+```
+
+### Backbone - required
+
+The `backbone` subsection is used to define the network, with all the network specific
+arguments under the same indent. The first argument should be the argument `name`, which
+should be string type, one of "unet", "local" or "global", to define a UNet, LocalNet or
+GlobalNet backbone, respectively. With Registry functionalities, you can also define
+your own networks to pass to DeepReg train via config.
+
+The `num_channel_initial` is used to define the number of initial channels for the
+network, and should be int type.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "unet" # One of unet, local, global: networks currently supported by DeepReg
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+```
+
+#### UNet
+
+The UNet model requires several additional arguments to define it's structure:
+
+- `depth`: int, defines the depth of the UNet from first to bottom, bottleneck layer.
+- `pooling`: Boolean, pooling method used for downsampling. True: non-parametrized
+  pooling will be used, False: conv3d will be used.
+- `concat_skip`: Boolean, concatenation method for skip layers in UNet. True:
+  concatenation of layers, False: addition is used instead.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "unet" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    depth: 3
+    pooling: false
+    concat_skip: true
+```
+
+#### Local and GlobalNet
+
+The LocalNet has an encoder-decoder structure, and extracts information from tensors at
+certain levels. We can define which levels to extract info from with the
+`extract_levels` argument.
+
+The GlobalNet encodes the image and uses the bottleneck layer to output an affine
+transformation using a FCN.
+
+- `extract_levels`: list of positive ints (ie, the min value in `extract_levels` should
+  be >=0). Eg. [1, 2, 3, 5] will extract information at those levels (but not 4).
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
```

Comment:
"global" net does not support extract_levels - worth adding a comment after this line

Reply:
Are you sure? https://github.com/DeepRegNet/DeepReg/blob/859df2a91564d16c43ea761ad272de568b8784bb/deepreg/model/backbone/global_net.py#L30

Reply:
@YipengHu

Reply:
> Are you sure?
> 
> https://github.com/DeepRegNet/DeepReg/blob/859df2a91564d16c43ea761ad272de568b8784bb/deepreg/model/backbone/global_net.py#L30

yup - the global net predicts parameters of transformation models, such as affine, without decoder. so this is redundant @mathpluscode  a ticket please

Reply:
Ok regardless of the fact of it being redundant or not... What to do with the docs? @YipengHu @mathpluscode

Reply:
good question. for this PR, we can only say "this is not used for global net, will be deprecated in future release" (and remember to do it...)

Reply:
Tickeeeet